sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveMetastoreCatalog.scala

```diff
@@ -277,6 +278,19 @@ private[hive] class HiveMetastoreCatalog(sparkSession: SparkSession) extends Log
     result.copy(output = newOutput)
   }
 
+  private def getDirectoryPathSeq(rootPath: Path): Seq[String] = {
+    val enableSupportSubDirectories =
+      sparkSession.sparkContext.
+        hadoopConfiguration.get("hive.mapred.supports.subdirectories", "false")
```

Comment:
This would fit even better:
https://hadoop.apache.org/docs/current/api/org/apache/hadoop/conf/Configuration.html#getBoolean-java.lang.String-boolean-